src/utils/findGitmojiCommand.ts

```diff
@@ -0,0 +1,86 @@
+// @flow
+import { CommitModes } from '@constants/commit'; // Assuming COMMIT_MODES is exported as CommitModes
+import FLAGS from '@constants/flags';
+
+interface Options {
+  [key: string]: any;
+}
+
+interface Cli {
+  flags: { [key: string]: any };
+  input: string[];
+  showHelp(): void;
+}
+
+const isSupportedCommand = (command: string | undefined, options: Options): boolean => {
+  return command !== undefined && Object.keys(options).includes(command);
+};
+
+type CommandType = 'flag' | 'command';
+
+interface CommandResult {
+  type: CommandType;
+  command: string | null;
+}
+
+const determineCommand = (flags: Options, input: string[], options: Options): CommandResult => {
+  const command = Object.keys(flags)
+    .map((flag) => flags[flag] && flag)
+    .find((flag) => options[flag]);
+
+  return command
+    ? {
+      type: 'flag',
+      command
+    }
+    : {
+      type: 'command',
+      command: input[0] || null
+    };
+};
+
+interface CommandOptions {
+  message?: string;
+  mode?: CommitModes;
+  scope?: string;
+  title?: string;
+  query?: string[];
+}
+
+const getOptionsForCommand = (
+  command: string | null,
+  flags: Options,
+  input: string[],
+  type: CommandType
+): CommandOptions | null => {
+  switch (command) {
+    case FLAGS.COMMIT:
+    case FLAGS.HOOK:
+      return {
+        message: flags['message'],
+        mode: command === FLAGS.HOOK ? CommitModes.HOOK : CommitModes.CLIENT,
+        scope: flags['scope'],
+        title: flags['title']
+      };
+    case FLAGS.SEARCH:
+      return {
+        query: type === 'command' ? input.slice(1) : input
+      };
+  }
+
+  return null;
+};
```

Comment:
While not wrong, wouldn't it make more sense to know which exact options you're getting back based on the command? Instead of just an object where everything is potentially `undefined`.

```suggestion
interface CommandOptions {
  message?: string;
  mode?: CommitModes;
  scope?: string;
  title?: string;
  query?: string[];
}

const getOptionsForCommand = (
  command: string | null,
  flags: Options,
  input: string[],
  type: CommandType
): CommandOptions | null => {
  switch (command) {
    case FLAGS.COMMIT:
    case FLAGS.HOOK:
      return {
        message: flags['message'],
        mode: command === FLAGS.HOOK ? CommitModes.HOOK : CommitModes.CLIENT,
        scope: flags['scope'],
        title: flags['title']
      };
    case FLAGS.SEARCH:
      return {
        query: type === 'command' ? input.slice(1) : input
      };
  }

  return null;
};
```